porcupine/plugins/directory_tree.py

```diff
@@ -459,8 +472,10 @@ def setup() -> None:
             tree.add_project(path, refresh=False)
     tree.refresh()
 
-    # TODO: invoking context menu from keyboard
+    # Invoke context menu from right-click
     tree.bind("<<RightClick>>", tree._on_right_click, add=True)
+    # Invoke context menu from menu key
+    tree.bind("<App>", tree._on_right_click, add=True)
```

Comment:
The correct key binding seems to be platform specific. On my computer (Linux), I get:

```
porcupine.pluginloader ERROR: directory_tree.setup() doesn't work
Traceback (most recent call last):
  File "/home/akuli/porcu/porcupine/pluginloader.py", line 190, in _run_setup_and_set_status
    info.module.setup()
  File "/home/akuli/porcu/porcupine/plugins/directory_tree.py", line 478, in setup
    tree.bind("<App>", tree._on_right_click, add=True)
  File "/usr/lib/python3.9/tkinter/__init__.py", line 1392, in bind
    return self._bind(('bind', self._w), sequence, func, add)
  File "/usr/lib/python3.9/tkinter/__init__.py", line 1346, in _bind
    self.tk.call(what + (sequence, cmd))
_tkinter.TclError: bad event type or keysym "App"
```

and the correct binding seems to be `<Menu>` instead of `<App>`. 

Does `<Menu>` work on your computer? If it doesn't, we can define a platform-specific virtual event in `porcupine/default_keybindings.tcl`, just like how `<<RightClick>>` is defined.

Reply:
I use Windows 10, `<App>` works but not `<Menu>`. I'm unfamiliar with virtual events, but I tried the following:
```
# Use Command on mac, Control on other systems
if {[tk windowingsystem] == "aqua"} {
    set contmand Command
    event add "<<RightClick>>" <Button-2>
    event add "<<RightClick>>" <Control-Button-1>
    event add "<<WheelClick>>" <Button-3>
} else {
    set contmand Control
    event add "<<RightClick>>" <Button-3>
    event add "<<WheelClick>>" <Button-2>
    event add "<<MenuKey>>" <App>                             <--- ADDED
}
```
with 

```suggestion
    tree.bind("<<MenuKey>>", tree._on_right_click, add=True)
```
And the feature continued to work on my computer. 
I could use help on where and how to correctly define the event on other platforms.

Reply:
Yeah, the Tcl code is a bit hard to read :)

The first line `if {[tk windowingsystem] == "aqua"} {` checks if Porcupine is running on MacOS. There are 3 possible values that `[tk windowingsystem]` can return: `aqua` (macos), `x11` (linux) and `win32` (windows). It's documented here: https://www.tcl.tk/man/tcl/TkCmd/tk.html#M12

So this if,else is for MacOS specific things. The `<App>` vs `<Menu>` difference isn't a MacOS thing though, because it differs between Windows and Linux. I put "macos keyboard" to google images search, and there doesn't seem to be a menu key on Mac.

I think we should just add a new if statement below, something like this:

```tcl
if {[tk windowingsystem] == "win32"} {
    event add "<<MenuKey>>" <App>
} elseif {[tk windowingsystem] == "x11"} {
    event add "<<MenuKey>>" <Menu>
}
```

Also, you aren't the first contributor who struggles with this Tcl code. Feel free to add a couple short comments to explain it for non-Tcl programmers.

Reply:
> I'm unfamiliar with virtual events

I have explained them here: https://akuli.github.io/porcupine/plugin-intro.html#virtual-events But that documentation is a bit outdated, because it doesn't mention that `default_keybindings.tcl` associates many virtual events with physical events (i.e. non-virtual events like `<App>` or `<Button-2>`).

Feel free to make a pull request that updates the documentation, or an issue that reminds me to update it. The documentation is somewhere in `docs/`. To find it, you can e.g. search for its title "Virtual Events" in all files, e.g. `git grep "Virtual Events"`.

Reply:
> Feel free to make a pull request that updates the documentation, or an issue that reminds me to update it.

Okay, I've made the issue #1028 for this.

TYSM for all your help! LMK about any other issues.